frame/composable-traits/src/financial_nft.rs

```diff
@@ -0,0 +1,150 @@
+use codec::{Decode, Encode};
+use frame_support::traits::{
+	tokens::nonfungibles::{Create, Inspect, Mutate},
+	Get,
+};
+use sp_runtime::{DispatchError, TokenError};
+
+pub trait FinancialNFTProvider<AccountId>: Create<AccountId> + Mutate<AccountId> {
+	///
+	/// Mint an NFT instance with initial (key, value) attribute in the given account.
+	///
+	/// Arguments
+	///
+	/// * `class` the NFT class id.
+	/// * `who` the owner of the minted NFT.
+	/// * `key` the key of the initial attribute.
+	/// * `value` the value of the initial attribute.
+	///
+	/// Returns the unique instance id.
+	fn mint_nft<K: Encode, V: Encode>(
+		class: &Self::ClassId,
+		who: &AccountId,
+		key: &K,
```

Comment:
Vec<(K,V))> ?

Reply:
The idea is not to store a list of attribute but directly the scale encoded struct under a single key, this function allow us to have the `mint_protocol_nft` that directly provide the NFT struct as single attribute (class, nft)